### PR TITLE
[Runtime] Allow casts from AnyHashable to a Hashable enum to succeed.

### PR DESCRIPTION
**Pull request by mikeash**
rdar://problem/46472361

### CODE REVIEW

**Comment by mikeash** [comment]
@swift-ci please test

**Comment by mikeash** [comment]
@swift-ci please test

**Review by rjmccall — APPROVED** [review]
LGTM.

File: stdlib/public/runtime/Casting.cpp [hunk 1]
```diff
@@ -2437,6 +2437,15 @@ static bool swift_dynamicCastImpl(OpaqueValue *dest, OpaqueValue *src,
                                           cast<StructMetadata>(targetType),
                                           flags);
       }
+
+      // AnyHashable casts for enums.
+      if (targetType->getKind() == MetadataKind::Enum) {
```
[the remark this answers is not in the document]
Comment (mikeash) [reply stdlib/public/runtime/Casting.cpp]: Thanks. I was trying to be explicit but it ended up a bit awkward. The comment alone is sufficient.

**Comment by mikeash** [comment]
@swift-ci please smoke test and merge

**Comment by mikeash** [comment]
@swift-ci please smoke test and merge

**Pull request merged**